diff --git a/node_modules/@cocalc/fuse-native/.bun-tag-c4d7362e03e4e6a9 b/.bun-tag-c4d7362e03e4e6a9
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/index.d.ts b/index.d.ts
index d89b13b6fcc81aa9f4db9940b430e62265c546b5..f237df338e92777e27101a13cab19266dc1a2174 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -144,6 +144,11 @@ declare namespace Fuse {
     modules?: string;
     name?: string;
     mnt?: string;
+
+    /**
+     * CUSTOM PATCH, TODO
+     */
+    volname?: string;
   }
 }
 
diff --git a/index.js b/index.js
index f051f8805a8729313dc870dd37c046fc22895f92..fd27f190018f2f2b4835c87930a0c73149099635 100644
--- a/index.js
+++ b/index.js
@@ -197,6 +197,10 @@ class Fuse extends Nanoresource {
       if (HAS_FOLDER_ICON) options.push('volicon=' + OSX_FOLDER_ICON)
     }
 
+    if (this.opts.volname) {
+      options.push('volname=' + this.opts.volname);
+    }
+
     return options.length ? '-o' + options.join(',') : ''
   }
 
